### PR TITLE
fix #41 as best as possible

### LINKED ISSUE #41: leaving spectator mode doesnt make you leave cinematicSpec
^title 

* scenario creator sends player into spec mode (either ACE or EG spec)
* player goes into cinematicSpec which switches off the spec mode
* scenario creator wants to pull  player from spec
    * spectator termination does not work because as far as the spectator mode is concerned, the player is *not* spectating

only practical solution is probably the mission creator manually also killing the cinematicSpec when trying to terminate spectator

### PR DESCRIPTION
**Pull request by Fusselwurm**
* add warning to cinematicSpec about possible breakage if player is pulled from spec while in cinematic
* add callbacked API function to pull players from cinematicSpec
* add API function to assure cinematicSpec about compatibility

### CODE REVIEW

**Comment by Fusselwurm** [comment]
As it's still in optionals, I'll just merge this. On moving to addons/, I'll keep the PR open for a bit longer.

File: optionals/cinematicSpec/functions/fn_api_cancelCinematicSpec.sqf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * [{
+ *     [false] call ace_spectator_fnc_setSpectator;
+ * }] call grad_cinematicSpec_fnc_cancelCinematicSpec;
+ *
+ * returns "sync" on immediate callback execution and "async" if callback execution is delayed
+ */
+
+#include "script_component.hpp"
+
+private _callback = param [0, {}];
```
Comment (dedmen) [inline optionals/cinematicSpec/functions/fn_api_cancelCinematicSpec.sqf]: Params? :u

File: optionals/cinematicSpec/functions/fn_api_cancelCinematicSpec.sqf [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+GRAD_CINEMACAM = objNull;
+
+[
+    {
+        isNil QUOTE(GRAD_CINEMACAM)
+    },
+    {
+        [
+            param [0],
```
Comment (dedmen) [inline optionals/cinematicSpec/functions/fn_api_cancelCinematicSpec.sqf]: add `,1` to L34 and just pass _this here?

File: optionals/cinematicSpec/functions/fn_api_cancelCinematicSpec.sqf (hunk 1 above)
Comment (Fusselwurm) [reply optionals/cinematicSpec/functions/fn_api_cancelCinematicSpec.sqf]: I *think* I'm not using params becaus @kayler-renslow's IntelliJ plugin wont recognize the variable declarations.

Comment (dedmen) [reply optionals/cinematicSpec/functions/fn_api_cancelCinematicSpec.sqf]: und schneller :u

Comment (chris579) [reply optionals/cinematicSpec/functions/fn_api_cancelCinematicSpec.sqf]: und macht alle params automatisch private

**Pull request merged**